<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <requires lib="gtk+" version="3.20"/>
  <template class="CcKeyboardShortcutEditor" parent="GtkDialog">
    <property name="can_focus">False</property>
    <property name="resizable">False</property>
    <property name="modal">True</property>
    <property name="width_request">400</property>
    <property name="height_request">300</property>
    <property name="window_position">center</property>
    <property name="type_hint">dialog</property>
    <signal name="delete-event" handler="gtk_widget_hide_on_delete" object="CcKeyboardShortcutEditor" swapped="yes"/>
    <child internal-child="vbox">
      <object class="GtkBox">
        <property name="can_focus">False</property>
        <property name="orientation">vertical</property>
        <child>
          <object class="GtkStack" id="stack">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="hexpand">True</property>
            <property name="vexpand">True</property>
            <property name="border_width">12</property>
            <child>
              <object class="GtkBox">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="orientation">vertical</property>
                <property name="spacing">18</property>
                <child>
                  <object class="GtkLabel" id="top_info_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="wrap">True</property>
                    <property name="wrap_mode">word-char</property>
                    <property name="width_chars">15</property>
                    <property name="max_width_chars">20</property>
                  </object>
                </child>
                <child>
                  <object class="GtkGrid" id="shortcut_grid">
                    <property name="can_focus">False</property>
                    <property name="row_spacing">12</property>
                    <property name="column_spacing">24</property>
                    <child>
                      <object class="GtkLabel">
                        <property name="visible">True</property>
                        <property name="can_focus">False</property>
                        <property name="hexpand">True</property>
                        <property name="label" translatable="yes">Current Shortcut</property>
                        <attributes>
                          <attribute name="weight" value="bold" />
                        </attributes>
                      </object>
                    </child>
                    <child>
                      <object class="GtkShortcutLabel" id="shortcut_accel_label">
                        <property name="visible">True</property>
                        <property name="can_focus">False</property>
                        <property name="halign">center</property>
                        <property name="disabled-text" translatable="yes">Disabled</property>
                      </object>
                      <packing>
                        <property name="top_attach">1</property>
                        <property name="left_attach">0</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkLabel">
                        <property name="visible">True</property>
                        <property name="can_focus">False</property>
                        <property name="hexpand">True</property>
                        <property name="label" translatable="yes">New Shortcut</property>
                        <attributes>
                          <attribute name="weight" value="bold" />
                        </attributes>
                      </object>
                      <packing>
                        <property name="top_attach">0</property>
                        <property name="left_attach">1</property>
                      </packing>
                    </child>
                    <child>
                      <object class="GtkShortcutLabel" id="new_shortcut_accel_label">
                        <property name="visible">True</property>
                        <property name="can_focus">False</property>
                        <property name="halign">center</property>
                        <property name="disabled-text" translatable="yes">Disabled</property>
                      </object>
                      <packing>
                        <property name="top_attach">1</property>
                        <property name="left_attach">1</property>
                      </packing>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkLabel" id="shortcut_conflict_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="hexpand">True</property>
                    <property name="wrap">True</property>
                    <property name="wrap_mode">word-char</property>
                    <property name="width_chars">15</property>
                    <property name="max_width_chars">20</property>
                    <property name="xalign">0</property>
                  </object>
                </child>
                <child>
                  <object class="GtkButton" id="reset_button">
                    <property name="label" translatable="yes">Reset</property>
                    <property name="visible">True</property>
                    <property name="can_focus">True</property>
                    <property name="receives_default">True</property>
                    <property name="halign">end</property>
                    <signal name="clicked" handler="reset_item_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                  </object>
                </child>
              </object>
              <packing>
                <property name="name">edit</property>
              </packing>
            </child>
            <child>
              <object class="GtkGrid">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="hexpand">True</property>
                <property name="vexpand">True</property>
                <property name="row_spacing">12</property>
                <property name="column_spacing">12</property>
                <child>
                  <object class="GtkLabel">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="label" translatable="yes">Name</property>
                    <property name="xalign">1</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">0</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkLabel">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="label" translatable="yes">Command</property>
                    <property name="xalign">1</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">1</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkLabel">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="label" translatable="yes">Shortcut</property>
                    <property name="xalign">1</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkLabel" id="new_shortcut_conflict_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="hexpand">True</property>
                    <property name="wrap">True</property>
                    <property name="wrap_mode">word-char</property>
                    <property name="width_chars">15</property>
                    <property name="max_width_chars">20</property>
                    <property name="xalign">0</property>
                  </object>
                  <packing>
                    <property name="left_attach">0</property>
                    <property name="top_attach">3</property>
                    <property name="width">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkEntry" id="name_entry">
                    <property name="visible">True</property>
                    <property name="can_focus">True</property>
                    <property name="hexpand">True</property>
                    <property name="sensitive" bind-source="edit_button" bind-property="active" bind-flags="default|invert-boolean" />
                    <signal name="notify::text" handler="name_entry_changed_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                  </object>
                  <packing>
                    <property name="left_attach">1</property>
                    <property name="top_attach">0</property>
                    <property name="width">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkEntry" id="command_entry">
                    <property name="visible">True</property>
                    <property name="can_focus">True</property>
                    <property name="hexpand">True</property>
                    <property name="sensitive" bind-source="edit_button" bind-property="active" bind-flags="default|invert-boolean" />
                    <signal name="notify::text" handler="command_entry_changed_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                  </object>
                  <packing>
                    <property name="left_attach">1</property>
                    <property name="top_attach">1</property>
                    <property name="width">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkToggleButton" id="edit_button">
                    <property name="label" translatable="yes">Edit</property>
                    <property name="visible">True</property>
                    <property name="can_focus">True</property>
                    <property name="receives_default">True</property>
                    <signal name="notify::active" handler="edit_custom_shortcut_button_toggled_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                  </object>
                  <packing>
                    <property name="left_attach">2</property>
                    <property name="top_attach">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkShortcutLabel" id="custom_shortcut_accel_label">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="halign">start</property>
                    <property name="hexpand">True</property>
                    <property name="disabled-text" translatable="yes">None</property>
                  </object>
                  <packing>
                    <property name="left_attach">1</property>
                    <property name="top_attach">2</property>
                  </packing>
                </child>
                <child>
                  <object class="GtkButton" id="remove_button">
                    <property name="visible">True</property>
                    <property name="can_focus">False</property>
                    <property name="label" translatable="yes">Remove</property>
                    <property name="valign">end</property>
                    <property name="sensitive" bind-source="edit_button" bind-property="active" bind-flags="default|invert-boolean" />
                    <signal name="clicked" handler="remove_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
                    <style>
                      <class name="destructive-action" />
                    </style>
                  </object>
                  <packing>
                    <property name="left_attach">2</property>
                    <property name="top_attach">3</property>
                  </packing>
                </child>
              </object>
              <packing>
                <property name="name">custom</property>
                <property name="position">1</property>
              </packing>
            </child>
          </object>
          <packing>
            <property name="expand">False</property>
            <property name="fill">True</property>
            <property name="position">1</property>
          </packing>
        </child>
      </object>
    </child>
    <child type="titlebar">
      <object class="GtkHeaderBar" id="headerbar">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="show_close_button">True</property>
        <child>
          <object class="GtkButton" id="cancel_button">
            <property name="label" translatable="yes">Cancel</property>
            <property name="visible">True</property>
            <property name="can_focus">True</property>
            <property name="receives_default">True</property>
            <signal name="clicked" handler="cancel_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="no" />
          </object>
        </child>
        <child>
          <object class="GtkButton" id="add_button">
            <property name="label" translatable="yes">Add</property>
            <property name="visible">True</property>
            <property name="can_focus">True</property>
            <property name="receives_default">True</property>
            <signal name="clicked" handler="add_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
            <style>
              <class name="suggested-action" />
            </style>
          </object>
          <packing>
            <property name="pack_type">end</property>
            <property name="position">2</property>
          </packing>
        </child>
        <child>
          <object class="GtkButton" id="replace_button">
            <property name="label" translatable="yes">Replace</property>
            <property name="visible">True</property>
            <property name="can_focus">True</property>
            <property name="receives_default">True</property>
            <signal name="clicked" handler="replace_button_clicked_cb" object="CcKeyboardShortcutEditor" swapped="yes" />
          </object>
          <packing>
            <property name="pack_type">end</property>
            <property name="position">2</property>
          </packing>
        </child>
      </object>
    </child>
  </template>
  <object class="GtkSizeGroup">
    <widgets>
      <widget name="cancel_button"/>
      <widget name="add_button"/>
      <widget name="replace_button"/>
      <widget name="reset_button"/>
    </widgets>
  </object>
</interface>
